packages/tinacms/src/toolkit/form-builder/form-builder.tsx

```diff
@@ -111,6 +111,17 @@ export const FormBuilder: FC<FormBuilderProps> = ({
 
   const tinaForm = form.tinaForm
   const finalForm = form.tinaForm.finalForm
+  const schema: TinaSchema = cms.api.tina.schema
+
+  React.useEffect(() => {
+    const collection = schema.getCollectionByFullPath(tinaForm.relativePath)
+    if (collection?.ui?.beforeSubmit) {
+      tinaForm.beforeSubmit = (values: any) =>
+        collection.ui.beforeSubmit({ cms, form: tinaForm, values })
+    } else {
+      tinaForm.beforeSubmit = undefined
+    }
+  }, [tinaForm.relativePath])
```

Comment:
wasn't sure if the form is updated so put this in a useEffect to be safe. let me know if it should be pulled out